tests/rps_v2_test.rb

```diff
@@ -0,0 +1,50 @@
+require 'test_helper'
+require 'rps_v2'
+
+class RockPaperScissorsTest < Minitest::Test
+
+	def test_player_one_win
+		sample_game =  RockPaperScissorsGame.new
+		sample_game.set_p1_weapon("rock")
+		sample_game.set_p2_weapon("scissors")
+		@win_hash = {"paper" => ["scissors", "lizard"], "scissors" => ["rock", "spock"], "rock" => ["paper", "spock"], "lizard" => ["rock", "scissors"], "spock" => ["paper", "lizard"]}
```

Comment:
You can't set an object's instance variables like this. If you want to set or change the value of `sample_game`'s `@win_hash` property, **the only** way to do it is using a method in the `RockPaperScissorsGame` class that does that. There are _no_ exceptions to this.

If you don't have a method in the `RockPaperScissorsGame` that sets `@win_hash`, then you should make one.

In your case, you _do_ have such a method. Great! :+1: You would use that method here instead of `@win_hash = {"paper" => ["sc...`. Because, and I need to stress this: You can't set an object's instance variables except by using one of that object's methods. It's *impossible*.

But now the question is, **why** are you setting `@win_hash` here at all? I know I suggested it, but you found an alternative approach, which is to simply call your `set_win_hash` method from inside the `find_winner` method. Smart. But again, why bother with `@win_hash = {"paper" => ["scissors", "liz...` then?